src/backend/distributed/citus--5.2-2--5.2-3.sql

```diff
@@ -0,0 +1,61 @@
+SET search_path = 'pg_catalog';
+
+/* add pg_dist_node */
+CREATE TABLE citus.pg_dist_node(
+	nodeid int NOT NULL PRIMARY KEY,
+	groupid int NOT NULL,
+	nodename text NOT NULL,
+	nodeport int NOT NULL,
+	UNIQUE (nodename, nodeport)
+);
+
+CREATE SEQUENCE citus.pg_dist_groupid_seq
+	MINVALUE 1
+	MAXVALUE 4294967296;
+
+CREATE SEQUENCE citus.pg_dist_node_nodeid_seq
+	MINVALUE 1
+	MAXVALUE 4294967296;
+
+ALTER TABLE citus.pg_dist_node SET SCHEMA pg_catalog;
+ALTER SEQUENCE citus.pg_dist_groupid_seq SET SCHEMA pg_catalog;
+ALTER SEQUENCE citus.pg_dist_node_nodeid_seq SET SCHEMA pg_catalog;
+
+CREATE FUNCTION master_dist_node_cache_invalidate()
+	RETURNS trigger
+	LANGUAGE C
+	AS 'MODULE_PATHNAME', $$master_dist_node_cache_invalidate$$;
+COMMENT ON FUNCTION master_dist_node_cache_invalidate()
+	IS 'invalidate internal cache of nodes when pg_dist_nodes changes';
+CREATE TRIGGER dist_node_cache_invalidate
+    AFTER INSERT OR UPDATE OR DELETE
+    ON pg_catalog.pg_dist_node
+    FOR EACH ROW EXECUTE PROCEDURE master_dist_node_cache_invalidate();
+
+CREATE FUNCTION master_add_node(nodename text,
+								nodeport integer,
+								groupid integer DEFAULT 0)
```

Comment:
For simplicity, maybe remove the groupid parameter until we actually start using it.